pkg/apis/servicecatalog/types.go

```diff
@@ -232,14 +232,8 @@ type ClusterServiceClass struct {
 	Status ClusterServiceClassStatus
 }
 
-// ClusterServiceClassSpec represents details about a ClusterServicePlan
-type ClusterServiceClassSpec struct {
-	// ClusterServiceBrokerName is the reference to the Broker that provides this
-	// ClusterServiceClass.
-	//
-	// Immutable.
-	ClusterServiceBrokerName string
-
+// SharedServiceClassSpec represents details about a ServiceClass
+type SharedServiceClassSpec struct {
```

Comment:
I understand that you are using the word `shared` to mean objects shared between `ClusterServiceClass` and `ServiceClass`, but I am wondering if in a world where there is a `CluserServiceClass` and `ServiceClass`, could there also be something called `SharedServiceClass` as an object that has a link between the two? 

Obvious choices are:
- shared
- common
- base
- core

If we are working on this area, perhaps we also consolidate the overlapping Class/Plan spec properties into a class like, CoreServiceSpec (or SharedServiceSpec)

I fear not liking `Shared` is a bit of a bikeshed but as I read the code I constantly wonder what the CluserServiceClass is being shared with and that is the twitch I am commenting about...

Reply:
How about `CommonServiceClassSpecFields` - it's okay to get descriptive with the name here, users will never be exposed to this name.

cc @eriknelson @jeremyrickard

Reply:
@pmorie I'm not opposed to that, but we've already accepted the `SharedServicePlan` PR. Do we think this is important enough to course-correct master and then update these PRs?

Reply:
I vote common. I think @n3wscott opened a PR to do a global name change once we finish up. I think that would be an acceptable way to handle it.

Reply:
That sounds good; let's get this clearly communicated on the proposal and I'll update the respective PRs.